src/components/Form/Label.jsx

```diff
@@ -3,8 +3,8 @@ import T from 'prop-types'
 import classNames from '../../utils/classNames'
 
 const Label = ({ className, children, htmlFor, ...rest }) => (
-  <label htmlFor={htmlFor} className={classNames('c-label', className)} {...rest}>
-    { children }
+  <label id="c-label" htmlFor={htmlFor} className={classNames('c-label', className)} {...rest}>
```

Comment:
Hard coded `id`.  Label's don't need an `id` so just let it be optionally passed with `...rest`